docs/x509.rst

```diff
@@ -964,6 +967,15 @@ Exceptions
 
         Returns the OID.
 
+.. class:: UnsupportedGeneralNameType
+
+    This is raised when a certificate contains an unsupported general name
+    type in an extension.
+
+    .. attribute:: type
+
+        :type: :term:`text` or int
```

Comment:
Oof, not really a fan of this. This is persuasive to me that we should just use ints everywhere.